test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/Sorting/SortTests.cs

```diff
@@ -0,0 +1,202 @@
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<Document>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.OK);
+
+            responseDocument.ManyData.Should().HaveCount(3);
+            responseDocument.ManyData[0].Id.Should().Be(articles[1].StringId);
+            responseDocument.ManyData[1].Id.Should().Be(articles[0].StringId);
+            responseDocument.ManyData[2].Id.Should().Be(articles[2].StringId);
+        }
+
+        [Fact]
+        public async Task Cannot_sort_in_single_primary_resource()
+        {
+            // Arrange
+            var article = new Article
+            {
+                Caption = "X"
+            };
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.GetCollection<Article>(nameof(Article)).InsertOneAsync(article);
+            });
+
+            var route = $"/api/v1/articles/{article.StringId}?sort=id";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<ErrorDocument>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.BadRequest);
+
+            responseDocument.Errors.Should().HaveCount(1);
+            responseDocument.Errors[0].StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            responseDocument.Errors[0].Title.Should().Be("The specified sort is invalid.");
+            responseDocument.Errors[0].Detail.Should().Be("This query string parameter can only be used on a collection of resources (not on a single resource).");
+            responseDocument.Errors[0].Source.Parameter.Should().Be("sort");
+        }
+
+        [Fact]
+        public async Task Cannot_sort_in_single_secondary_resource()
+        {
+            // Arrange
+            var article = new Article
+            {
+                Caption = "X"
+            };
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.GetCollection<Article>(nameof(Article)).InsertOneAsync(article);
+            });
+
+            var route = $"/api/v1/articles/{article.StringId}/author?sort=id";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<ErrorDocument>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.BadRequest);
+
+            responseDocument.Errors.Should().HaveCount(1);
+            responseDocument.Errors[0].StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            responseDocument.Errors[0].Title.Should().Be("The specified sort is invalid.");
+            responseDocument.Errors[0].Detail.Should().Be("This query string parameter can only be used on a collection of resources (not on a single resource).");
+            responseDocument.Errors[0].Source.Parameter.Should().Be("sort");
+        }
+
```

Comment:
Add the following missing tests:
- Can_sort_on_HasMany_relationship
- Can_sort_on_HasManyThrough_relationship
- Can_sort_on_HasOne_relationship

The difference is they are supposed to fail (assert on that)

Reply:
I'm still failing `Cannot_sort_on_HasMany_relationship` so consider this partially address for now.

Reply:
This test needs a bit more to bypass an optimization. The MongoDB collection is empty, so when `.CountAsync()` is called, no documents are returned, so the actual query to fetch data never executes, so the sort is never applied. The solution is to add a single document upfront and the test passes.

Reply:
I had copied these tests without the complete `Arrange` part, I mean, I had not added resources to the database prior to calling the endpoints. This worked for `HasOne` and `HasManyThrough` but it did not work for `HasMany`. Adding this part of the test made `HasMany` work too. Do you think we should do something about this?

I mean, should we expect an API to return `400` when the database is empty and we try to use relationships? I think we should but I don't know how I can do that neatly hehe